server/routes/basicDetails.ts

```diff
@@ -0,0 +1,411 @@
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.dateOfLetter = userEnteredDateOfLetter
+      errorMessages.dateOfLetter = {
+        text: 'The proposed date for this letter is in an invalid format, please use the correct format e.g 17/5/2024',
+      }
+      // we cant continue with date validation
+      return errorMessages
+    }
+
+    if (breachNotice) {
+      // check date of letter is not before today
+      if (breachNotice.dateOfLetter) {
+        const localDateOfLetterAtStartOfDay = LocalDate.parse(breachNotice.dateOfLetter).atStartOfDay()
+        if (localDateOfLetterAtStartOfDay.isBefore(currentDateAtStartOfTheDay)) {
+          errorMessages.dateOfLetter = {
+            text: 'The letter has not been completed and so the date cannot be before today.',
+          }
+        }
+        if (localDateOfLetterAtStartOfDay.minusDays(7).isAfter(currentDateAtStartOfTheDay)) {
+          errorMessages.dateOfLetter = {
+            text: 'The proposed date for this letter is a week in the future. Breach letters need to be timely in order to allow evidence to be presented. Please complete the letter in a timely way or update the contact outcome to be acceptable.',
+          }
+        }
+      }
+    }
+    return errorMessages
+  }
+
+  function getSelectedAddress(addressList: AddressList, addressIdentifier: string): Address {
+    const addressIdentifierNumber: number = +addressIdentifier
+    return addressList.find(address => address.addressId === addressIdentifierNumber)
+  }
+
+  async function renderWarningTypes(
+    breachNotice: BreachNotice,
+    res: Response,
+    warningTypeRadioButtons: Array<RadioButton>,
+    errorMessages: ErrorMessages,
+    sentenceTypeSelectItems: Array<SelectItem>,
+  ) {
+    res.render('pages/warning-type', {
+      errorMessages,
+      breachNotice,
+      warningTypeRadioButtons,
+      sentenceTypeSelectItems,
+    })
+  }
+
+  router.get('/basic-details', async (req, res, next) => {
+    await auditService.logPageView(Page.BASIC_DETAILS, { who: res.locals.user.username, correlationId: req.id })
+    res.render('pages/basic-details')
+  })
+
+  // if this page hasnt been saved we want to go through and apply defaults otherwise dont do this
+  function checkBreachNoticeAndApplyDefaults(breachNotice: BreachNotice, basicDetails: BasicDetails) {
+    // we havent saved the page yet so default
+    if (!breachNotice.basicDetailsSaved) {
+      // load offender name from integration details
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.titleAndFullName = `${basicDetails.title} ${basicDetails.name.forename} ${basicDetails.name.middleName} ${basicDetails.name.surname}`
+      // default the radio buttons to true
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.useDefaultAddress = true
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.useDefaultReplyAddress = true
+    }
+  }
```

Comment:
Could we keep breachNotice immutable here by returning a new one instead of updating in-place?  That would get rid of all the eslint errors.

```suggestion
  function applyDefaults(breachNotice: BreachNotice, basicDetails: BasicDetails) {
    return { 
      ...breachNotice, 
      titleAndFullName: `${basicDetails.title} ${basicDetails.name.forename} ${basicDetails.name.middleName} ${basicDetails.name.surname}`,
      useDefaultAddress: true,
      useDefaultReplyAddress: true
    }
  }
```